src/Runner.Worker/FileCommandManager.cs

```diff
@@ -259,4 +259,99 @@ public void ProcessCommand(IExecutionContext context, string filePath, Container
             return text.Substring(originalIndex, lfIndex - originalIndex);
         }
     }
+
+    public sealed class SetStepSummaryCommand : RunnerService, IFileCommandExtension
+    {
+        public string ContextName => "step_summary";
+        public string FilePrefix => "step_summary_";
+
+        private const int AttachmentSizeLimit = 128 * 1024;
+
+        public Type ExtensionType => typeof(IFileCommandExtension);
+
+        public void ProcessCommand(IExecutionContext context, string filePath, ContainerInfo container)
+        {
+            if (ShouldUploadAttachment(context, filePath))
+            {
+                Trace.Info($"Submitting step summary content from file {filePath}, container: {container}");
+                var scrubbedFilePath = ScrubStepSummaryFileSecrets(filePath);
+
+                QueueStepSummaryUpload(context, scrubbedFilePath);
+            }
+        }
+
+        private bool ShouldUploadAttachment(IExecutionContext context, string filePath)
+        {
+            if (!ActionCommandManager.StepSummaryEnabled(context))
+            {
+                Trace.Info("Step Summary is disabled; skipping attachment upload");
+                return false;
+            }
+
+            if (String.IsNullOrEmpty(filePath))
```

Comment:
i would combine all file exit/empty checks into a single `if` and trace.

Reply:
@TingluoHuang I found the separate trace statements useful when testing/debugging. I guess now that I have unit tests for all of them, they might not be as useful any more. Is that what you're aiming at?

Reply:
we can revisit this when you inline all the methods. my main goal is to get the minimal key tracing, ex: `we don't upload the summary because it's not there or empty.`